R/delineate.R

```diff
@@ -0,0 +1,53 @@
+#' Delineate a corridor around a river.
+#'
+#' @param city_name A place name as a string
+#' @param river_name A river name as a string
+#' @param crs The projected Coordinate Reference System (CRS) to use. If not
+#'   provided, the suitable Universal Transverse Mercator (UTM) CRS is selected
+#' @param bbox_buffer Add a buffer region to the city boundary to avoid edge
+#'   effects close to its limits
+#' @param initial_method The method employed to define the initial river
+#'   corridor geometry. See [initial_corridor()] for the available methods
+#' @param capping_method The method employed to connect the corridor edge end
+#'   points (i.e. to "cap" the corridor). See [cap_corridor()] for
+#'   the available methods
+#' @param segments Whether to carry out the corridor segmentation
+#' @param riverspace Whether to carry out the riverspace delineation
+#'
+#' @return A simple feature geometry
+#' @export
+delineate_corridor <- function(
+  city_name, river_name, crs = NULL, bbox_buffer = NULL,
+  initial_method = "buffer", capping_method = "direct", segments = FALSE,
+  riverspace = FALSE
+) {
+  # Retrieve all relevant OSM datasets using the extended bounding box
+  osm_data <- get_osmdata(city_name, river_name, crs, bbox_buffer)
+
+  # Define the area of interest
+  bbox <- sf::st_transform(osm_data$bb, sf::st_crs(osm_data$boundary))
```

Comment:
```suggestion

  # Define the area of interest
  bbox <- osm_data$bb
```
I do not remember why `bb` does not have the same CRS as the other layers included in the `osm_data` list, so I updated `get_osmdata()` to transform bb to the same projected CRS.